malariagen_data/anopheles.py

```diff
@@ -7726,6 +7727,591 @@ def plot_h1x_gwss(
+            Minimum EHH beyond which to truncate integrated haplotype homozygosity
+            calculation.
+        max_gap : int, optional
+            Do not report scores if EHH spans a gap larger than this number of base pairs
+        gap_scale : int, optional
+            Rescale distance between variants if gap is larger than this value
+        include_edges : bool, optional
+            If True, report scores even if EHH does not decay below min_ehh at the
+            end of the chromosome.
+        use_threads : bool, optional
+            If True, use multiple threads to compute iHS.
+        min_cohort_size : int, optional
+            If provided, raise a ValueError if the number of samples in the cohort is
+            less than this value.
+        max_cohort_size : int, optional
+            If provided, randomly down-sample to this value if the number of
+            samples in the cohort is greater.
+        random_seed : int, optional
+            Random seed used for down-sampling.
+
+        Returns
+        -------
+        x : numpy.ndarray
+            An array containing the window centre point genomic positions.
+        ihs : numpy.ndarray
+            An array with iHS statistic values for each window.
+
+        """
+        # change this name if you ever change the behaviour of this function, to
+        # invalidate any previously cached data
+        name = self._ihs_gwss_cache_name
+
+        params = dict(
+            contig=contig,
+            analysis=self._prep_phasing_analysis_param(analysis=analysis),
+            window_size=window_size,
+            percentiles=percentiles,
+            standardize=standardize,
+            standardization_bins=standardization_bins,
+            standardization_n_bins=standardization_n_bins,
+            standardization_diagnostics=standardization_diagnostics,
+            filter_min_maf=filter_min_maf,
+            compute_min_maf=compute_min_maf,
+            min_ehh=min_ehh,
+            include_edges=include_edges,
+            max_gap=max_gap,
+            gap_scale=gap_scale,
+            use_threads=use_threads,
+            sample_sets=self._prep_sample_sets_param(sample_sets=sample_sets),
+            # N.B., do not be tempted to convert this sample query into integer
+            # indices using _prep_sample_selection_params, because the indices
+            # are different in the haplotype data.
+            sample_query=sample_query,
+            min_cohort_size=min_cohort_size,
+            max_cohort_size=max_cohort_size,
+            random_seed=random_seed,
+        )
+
+        try:
+            results = self.results_cache_get(name=name, params=params)
+
+        except CacheMiss:
+            results = self._ihs_gwss(**params)
+            self.results_cache_set(name=name, params=params, results=results)
+
+        x = results["x"]
+        ihs = results["ihs"]
+
+        return x, ihs
+
+    def _ihs_gwss(
+        self,
+        contig,
```

Comment:
```suggestion
    def _ihs_gwss(
        self,
        *,
        contig,
```

force use of keyword arguments, prevents bugs due to args in wrong order